ukmm: 0.11.1 -> 0.12.1

kira-bruneau opened the pull request:
## Description of changes

https://github.com/NiceneNerd/UKMM/releases/tag/v0.12.1

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#reviewing-contributions
-->

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

kira-bruneau committed:
ukmm: 0.11.1 -> 0.12.1

symphorien commented:
fails to run for me (I'm on x11)
```
$  results/ukmm/bin/ukmm 
thread 'main' panicked at /build/cargo-vendor-dir/xkbcommon-dl-0.4.2/src/x11.rs:59:28:
Library libxkbcommon-x11.so could not be loaded.
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
An unrecoverable error occured. Error details: Library libxkbcommon-x11.so could not be loaded.
```

kira-bruneau committed:
ukmm: fix running on x11

Also switches from using LD_LIBRARY_PATH to RUSTFLAGS to include
required runtime libraries.

kira-bruneau commented:
Oh thanks for the heads up! I haven't been testing this on X11. It looks like the dependency was added in 0.10.1 -> 0.11.0: https://github.com/NiceneNerd/UKMM/pull/166/files#diff-13ee4b2252c9e516a0547f2891aa2105c3ca71c6d7a1e682c69be97998dfc87eR5764-R5775

kira-bruneau commented:
Sorry the first force push was from the wrong base commit and left off the update commit - it should be good now though!

symphorien commented on pkgs/tools/games/ukmm/default.nix:
```diff
@@ -36,7 +37,12 @@ rustPlatform.buildRustPackage rec {
 
   preFixup = ''
-    gappsWrapperArgs+=(--prefix LD_LIBRARY_PATH : ${lib.makeLibraryPath [ libglvnd ]})
+    gappsWrapperArgs+=(--prefix LD_LIBRARY_PATH : ${
```
LD_LIBRARY_PATH leaks into child processes (ie firefox if you click on a link). this can have undesirable effects. Can you instead:
- force-link the libs with NIX_LDFLAGS ?
- add them to the rpath with patchelf ?

kira-bruneau replied:
This took me a little bit longer than I expected, apparently `NIX_LDFLAGS` doesn't change any of the rust link flags. I noticed that the wrapper executable linked to the libraries from `NIX_LDFLAGS` but not `.ukmm-wrapped`... so I copied over a pattern I saw other packages were using with `RUSTFLAGS`.

It should be good now! Thanks for the review! :blush:

symphorien commented:
thank you
